topaz-backend-main/main.py

```
@@ -1254,6 +1286,91 @@ async def save_user_metrics(metrics_request: UserMetricsRequest, request: Reques
+            return JSONResponse(status_code=503, content={"success": False, "error": "Database unavailable"})
+
+        user = getattr(request.state, "user", None)
+        if not user or not user.get("id"):
+            return JSONResponse(status_code=401, content={"success": False, "error": "Unauthorized"})
+
+        # Set per-request auth so RLS policies evaluate with the user's identity
+        try:
+            user_token = request.headers.get("X-User-Token")
+            if user_token and hasattr(supabase, "postgrest"):
+                supabase.postgrest.auth(user_token)
+        except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)